jekyll/_cci2/creating-orbs.md

```diff
@@ -0,0 +1,332 @@
+
+* Any member of an organization can publish dev orbs in namespaces.
+
+##### Development and Production Orb Retention and Mutability Characteristics
+
+* Dev orbs are mutable and expire. Anyone can overwrite any development orb who is a member of the organization that owns the namespace in which that orb is published.
+
+* Production orbs are immutable and long-lived. Once you publish a production orb at a given semantic version you may not change the content of that orb at that version. To change the content of a production orb you must publish a new version. CircleCI generally recommends using the ```orb publish increment``` and/or the ```orb publish promote``` commands in the ```circleci``` CLI when publishing orbs to production.
+
+##### Development and Production Orbs Versioning Semantics
+
+Development orbs are tagged with the format ```dev:<< your-string >>```. Production orbs are always published using the semantic versioning ("semver") scheme.
+
+In development orbs, the string label given by the user has the following restrictions:
+
+* Up to 1023 non-whitespace characters
+
+Examples of valid development orb tags:
+
+* Valid:
+  "dev:mybranch"
+  "dev:2018_09_01"
+  "dev:1.2.3-rc1"
+  "dev:myinitials/mybranch"
+  "dev:myVERYIMPORTANTbranch"
+
+* Invalid
+  "dev: 1" (No spaces allowed)
+  "1.2.3-rc1" (No leading "dev:")
+
+In production orbs you must use the form ```X.Y.Z``` where ```X``` is a "major" version, ```Y``` is a "minor" version, and ```Z``` is a "patch" version. For example, 2.4.0 implies the major version 2, minor version 4, and the patch version of 0.
+
+While not strictly enforced, CircleCI recommends when versioning your production orbs you use the standard semantic versioning convention for deciding what major, minor, and patch should mean semantically:
+
+* MAJOR: when you make incompatible API changes
+* MINOR: when you add functionality in a backwards-compatible manner
+* PATCH: when you make backwards-compatible bug fixes
+
+#### Using Orbs in Orbs and Register-Time Resolution
+
+You may use an ```orbs``` stanza inside an orb, providing a way to pull in other orbs' elements into your orb. The scoping rules are the same as orbs in configuration, and you declare inline orbs in an orb.
+
+<aside class="notice">
+Because production orb releases are immutable, the system will resolve all orb dependencies at the time you register your orb rather than the time you run your build (eager resolution).
+
+For example, if you choose to publish orb ```foo/bar``` at version ```1.2.3```, and that orb you are publishing has, internally, an ```orbs``` stanza that imports another orb referenced as ```biz/baz@volatile```, the latest version at the time you want to perform this action is ```2.1.0```. At the time you register ```foo/bar@1.2.3``` the system will resolve ```biz/baz@volatile``` as the latest version and include its elements directly into the packaged version of ```foo/bar@1.2.3```.
```

Comment:
This would be easier to read as a chronological event list, instead of in paragraph form.

Reply:
I simplified this to For example, orb ```foo/bar``` is published at version ```1.2.3``` with an ```orbs``` stanza that imports ```biz/baz@volatile```. At the time you register ```foo/bar@1.2.3``` the system will resolve ```biz/baz@volatile``` as the latest version and include its elements directly into the packaged version of ```foo/bar@1.2.3```.

If ```biz/baz``` is updated to ```3.0.0```, anyone using ```foo/bar@1.2.3``` will not see the change in ```biz/baz@3.0.0``` until ```foo/bar``` is published at a higher version than `1.2.3`.